nuxeo-ai-similar-content/src/main/java/org/nuxeo/ai/similar/content/operation/FindSimilar.java

```diff
@@ -0,0 +1,165 @@
+    protected String xpath = FILE_CONTENT;
+
+    @Param(name = "max", required = false)
+    protected long max;
+
+    @OperationMethod
+    public List<DocumentModel> run(DocumentModel doc) throws OperationException {
+        if (!scs.anyMatch(doc)) {
+            log.debug("None dedup filters fit document {}", doc.getId());
+            return emptyList();
+        }
+
+        List<String> ids = findSimilarIds(doc, null);
+        return resolveDocuments(ids);
+    }
+
+    @OperationMethod
+    public List<DocumentModel> run(Blob blob) throws OperationException {
+        if (blob.getLength() >= Long.parseLong(
+                Framework.getProperty(AI_BLOB_MAX_SIZE_CONF_VAR, AI_BLOB_MAX_SIZE_VALUE))) {
+            throw new OperationException("Blob is to large; size = " + blob.getLength());
+        }
+
+        List<String> ids = findSimilarIds(null, blob);
+        return resolveDocuments(ids);
+    }
+
+    protected List<String> findSimilarIds(@Nullable DocumentModel doc, @Nullable Blob blob) throws OperationException {
+        Optional<InsightClient> client = this.client.getClient(session);
+        if (client.isEmpty()) {
+            throw new OperationException(
+                    "Could not obtain Insight Client for user " + session.getPrincipal().getActingUser());
+        }
+
+        String id = doc == null ? null : doc.getId();
+        TensorInstances tensor = constructTensor((ManagedBlob) blob, id);
+
+        Map<String, Serializable> parameters = new HashMap<>();
+        parameters.put(UID, id);
+        parameters.put(XPATH_PARAM, xpath);
+
+        try {
+            InsightClient insight = client.get();
+            return insight.api(API.Dedup.FIND).call(parameters, tensor);
+        } catch (IOException e) {
+            throw new OperationException("Could not call Find API", e);
+        }
+    }
+
+    protected List<DocumentModel> resolveDocuments(List<String> ids) {
+        if (ids == null || ids.isEmpty()) {
+            log.debug("No similar documents found");
+            return emptyList();
+        }
+
+        List<DocumentRef> refs = ids.stream()
+                                    .map(IdRef::new)
+                                    .filter(ref -> session.exists(ref))
+                                    .collect(Collectors.toList());
+        if (refs.size() != ids.size()) {
+            log.warn("Deduplication found some nonexistent document, consider reindexing");
+        }
+
+        return session.getDocuments(refs.toArray(DocumentRef[]::new));
+    }
+
+    protected TensorInstances constructTensor(ManagedBlob blob, String id) {
+        if (blob == null) {
+            return null;
+        }
+
+        Map<String, TensorInstances.Tensor> props = new HashMap<>();
+        props.put(xpath, TensorInstances.Tensor.image(convertImageBlob(blob)));
+        return new TensorInstances(id, singletonList(props));
+    }
+
+    protected String convertImageBlob(ManagedBlob blob) {
```

Comment:
Something to discuss: Here we need to use the same conversion for both index/find operations. In my ticket we are taking about conversion like we did for the export `org.nuxeo.ai.pipes.functions.PropertyUtils#getPictureConversion`. But with different specifications which is having less than 5MB in size (no restriction in term of dimension). Something maybe to sync too.